builds/jenkins-build-sls-app/Jenkinsfile

```diff
@@ -0,0 +1,1062 @@
+    echo "prepareServerlessYml = ${dd}"
+    writeYaml(file: './serverless.yml', data: dd)
+
+    if (domain && domain == "jazz") {
+      stage('Update Service Template') {
+        echo "Inside Update service template"
+        try {
+          updatePlatformServiceTemplate(config, config['iamRoleARN'], roleId, current_environment, repo_name)
+        } catch (error) {
+          error "Error occured while updating service template."
+        }
+      }
+    }
+    stage('Pre-Build Validation') {
+      events.sendStartedEvent('VALIDATE_PRE_BUILD_CONF', 'pre-build validation started', context_map, environment_logical_id)
+      try {
+        send_status_email(config, 'STARTED', "")
+        validateDeploymentConfigurations(config)
+      } catch (ex) {
+        send_status_email(config, 'FAILED', "")
+        events.sendFailureEvent('VALIDATE_PRE_BUILD_CONF', ex.getMessage(), context_map, environment_logical_id)
+        events.sendFailureEvent('UPDATE_DEPLOYMENT', ex.getMessage(), environmentDeploymentMetadata.generateDeploymentMap("failed", environment_logical_id, gitCommitHash), environment_logical_id)
+        error ex.getMessage()
+      }
+      events.sendCompletedEvent('VALIDATE_PRE_BUILD_CONF', 'pre-build validation completed', context_map, environment_logical_id)
+    }
+
+    if (config['domain'] && config['domain'] != 'jazz' && configLoader.CODE_QUALITY.SONAR.ENABLE_SONAR == "true") {
+      stage('Code Quality Check') {
+        events.sendStartedEvent('CODE_QUALITY_CHECK', 'code quality check starts', context_map, environment_logical_id)
+        try {
+          runValidation(runtime)
+          sonarModule.doAnalysis()
+        } catch (ex) {
+          events.sendFailureEvent('CODE_QUALITY_CHECK', ex.getMessage(), context_map, environment_logical_id)
+          events.sendFailureEvent('UPDATE_DEPLOYMENT', ex.getMessage(), environmentDeploymentMetadata.generateDeploymentMap("failed", environment_logical_id, gitCommitHash), environment_logical_id)
+          error ex.getMessage()
+        }
+        events.sendCompletedEvent('CODE_QUALITY_CHECK', 'code quality check completed', context_map, environment_logical_id)
+      }
+    }
+
+    stage('Build') {
+      events.sendStartedEvent('BUILD', 'build starts', context_map, environment_logical_id)
+      try {
+        buildLambda(runtime, repo_name)
+      } catch (ex) {
+        events.sendFailureEvent('BUILD', ex.getMessage(), context_map, environment_logical_id)
+        events.sendFailureEvent('UPDATE_DEPLOYMENT', ex.getMessage(), environmentDeploymentMetadata.generateDeploymentMap("failed", environment_logical_id, gitCommitHash), environment_logical_id)
+        error ex.getMessage()
+      }
+      events.sendCompletedEvent('BUILD', 'build completed', context_map, environment_logical_id)
+    }
+
+    def env_key
+    if (branch == "master") {
+      env_key = "PROD"
+    } else {
+      env_key = "DEV"
+    }
+
+    stage('Package') {
+      echo 'The packaging state'
```

Comment:
Every  stage should send started, failed, success events in each stage.

These events should also be added to events datastore in [jazz-installer](https://github.com/tmobile/jazz-installer/blob/develop/installscripts/jazz-terraform-unix-noinstances/dynamodb-items-event-name.tf).

Reply:
jazz-installer repo seems missing the sls-app branch I need to clone in order to move my changes in. I'll speak to Rajeev about this tomorrow (Sun, Mar 17). May be it makes more sense if Rajeev handles this item as other install related artifact changes

Reply:
Resolved with the commit that changes jenkins-build-sls-app/Jenkinsfile and jazz-build-module/ events-module.groovy follows: 'Package' and 'Verify Template' send started, failed and success events. Adding the new event name into build-module

Reply:
Resolved with https://github.com/tmobile/jazz/pull/829/commits/ed93653c678f74bdc629c7932e2d8cc043e9e85f except install part due to absence of 'sls-app' branch in the repo

Reply:
JAZZ-INSTALLER part is addressed with the pull request as follows: https://github.com/tmobile/jazz-installer/pull/511